.autograding/__tests__/dom.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @jest-environment jsdom
+ */
+const { parseHTML, readFile } = require('../setup.js');
+const { dates } = require('../../src/js/dates.js');
+const html = readFile('../src/index.html');
+const doc = parseHTML(html);
+
+// HTML tests
+describe('DOM', function () {
+    beforeAll(() => {
+        document.documentElement.innerHTML = html.toString();
+        require('../../src/js/index.js');
+    });
+
+    afterAll(() => {
+        // restore the original func after test
+        jest.resetModules();
+    });
+
+    test('found timeline items', () => {
+        expect(document.getElementsByClassName('timeline-item').length).toBe(dates.length);
```

Comment:
Should probably make any tests like this more flexible and check greater or equal to the dates length. Any one of us could decide last min to make `dates.js` the minimum data or a student could decide to add dates too which would break the tests

Reply:
Yep, makes sense